p2p/transport/webrtc/udp_mux_conn.go

```diff
@@ -0,0 +1,246 @@
+package libp2pwebrtc
+
+import (
+	"encoding/binary"
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/pion/logging"
+	"github.com/pion/transport/packetio"
+)
+
+type udpMuxedConnParams struct {
+	Mux       *UDPMuxNewAddr
+	AddrPool  *sync.Pool
+	Key       string
+	LocalAddr net.Addr
+	Logger    logging.LeveledLogger
+}
+
+// udpMuxedConn represents a logical packet conn for a single remote as identified by ufrag
+type udpMuxedConn struct {
+	params *udpMuxedConnParams
+	// remote addresses that we have sent to on this conn
+	addresses []string
+
+	// channel holding incoming packets
+	buffer     *packetio.Buffer
+	closedChan chan struct{}
+	closeOnce  sync.Once
+	mu         sync.Mutex
+}
+
+func newUDPMuxedConn(params *udpMuxedConnParams) *udpMuxedConn {
+	p := &udpMuxedConn{
+		params:     params,
+		buffer:     packetio.NewBuffer(),
+		closedChan: make(chan struct{}),
+	}
+
+	return p
+}
+
+func (c *udpMuxedConn) ReadFrom(b []byte) (n int, raddr net.Addr, err error) {
```

Comment:
I'm not sure what you're actually reading here (this is most likely me being ignorant about specifics of the WebRTC encoding). Can you add a comment linking to the data format that's getting parsed here?

Reply:
https://github.com/libp2p/go-libp2p/pull/1655/files#diff-f175016b12b6cf30881b97cb5812f76ce03a6ef6370ccf3be2664c9114b8b96dR176

The data received on the shared socket is written here encoded as mentioned above.